packages/eslint-rules/src/rules/prefer_amplify_errors.ts

```diff
@@ -0,0 +1,46 @@
+import { ESLintUtils } from '@typescript-eslint/utils';
+import { join, sep } from 'path';
+
+export const preferAmplifyErrorsRule = ESLintUtils.RuleCreator.withoutDocs({
+  create(context) {
+    return {
+      // This naming comes from @typescript-eslint/utils types.
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      NewExpression(node) {
+        const checkNode = (errorType: string, ignorePatterns: string) => {
+          const fileNameWithPath = context.physicalFilename!.replace(
+            join(context.cwd, sep),
+            ''
+          );
+
+          const ignoreMatch = [
+            ...fileNameWithPath.matchAll(new RegExp(ignorePatterns, 'g')),
+          ];
+          if (
+            node.callee.type === 'Identifier' &&
+            node.callee.name === errorType &&
+            !ignoreMatch.length
+          ) {
+            context.report({
+              messageId: 'useOfErrorDetected',
+              node,
+            });
+          }
+        };
+        checkNode('Error', '.test.ts');
```

Comment:
I guess this was added for testing. should we remove it?

Btw. with these overrides in the other file - have you found a solution to filter out test files ?

Reply:
There is a way to filter for tests, I discussed a bit in the previous comment, I want to see if I can find a way to do it where I only have to have one list item that will filter out all tests. In my previous attempt I had to have list item for every directory level.

Reply:
Ok. I didn't realize how this parameter is used. If we keep regex this is fine.

Reply:
committing an update that filters out tests using the same method as filtering for packages we want the rules to be applied on